server/src/subscriber/subscriber.service.ts

```diff
@@ -40,22 +41,68 @@ export class SubscriberService {
   }
 
   async create(email: string, list: string, @Res() response) {
+    const id = crypto.randomUUID();
     const addRequest = {
       url: "/v3/marketing/contacts",
       method:<HttpMethod> 'PUT',
       body: {
         "list_ids": [list],
-        "contacts": [{"email": email, "anonymous_id": crypto.randomUUID()}]
+        "contacts": [{"email": email, "anonymous_id": id}]
       }
     }
 
     // If successful, this will add the request to the queue and return a 202
     // https://www.twilio.com/docs/sendgrid/api-reference/contacts/add-or-update-a-contact
     try {
       const result = await this.client.request(addRequest);
-      return {isError: false, result: result};
+      return {isError: false, result: result, anonymous_id: id};
     } catch(error) {
-      return {isError: true, ...error};
+      console.error(error, addRequest.body)
+      Sentry.captureException({error, email, id, list})
+      return {isError: true, ...error, request_body: addRequest.body};
     }
   }
+
+  async checkCreate(importId: string, @Res() response, counter: number = 0, checksBeforeFail: number, pauseBetweenChecks: number, list: string, errorInfo: any) {
+    if(counter >= checksBeforeFail) {
+      console.error({
+        code: 408,
+        message: `Polling limit of ${checksBeforeFail} checks with a ${pauseBetweenChecks/1000} second delay between each has been reached.`,
+        job_id: importId,
+        errorInfo
+      })
+      Sentry.captureException({
+        code: 408,
+        message: `Polling limit of ${checksBeforeFail} checks with a ${pauseBetweenChecks/1000} second delay between each has been reached.`,
+        job_id: importId,
+        errorInfo
+      })
+      return { isError: true, code: 408, errorInfo }
+    }
+
+    await delay(pauseBetweenChecks);
+
+    const confirmationRequest = {
+      url: `/v3/marketing/contacts/imports/${importId}`,
+      method:<HttpMethod> 'GET',
+    }
+
+    // https://www.twilio.com/docs/sendgrid/api-reference/contacts/import-contacts-status
+    try {
+      const user = await this.client.request(confirmationRequest);
```

Comment:
Can you change this to `job` or something similar? This endpoint gives you back details about the request to import a contact, not an object representing an actual user, so this was confusing to me as someone coming to the code for the first time.